measurement/util.py

```diff
@@ -30,9 +31,17 @@ def cwd():
     return os.getcwd()
 
 
-def run_cmd(cmd):
-    p = Popen(cmd, stdout=PIPE, stderr=PIPE)
-    return p.communicate()
+def run_cmd(cmd, shell=False, timeout=None):
+    if python_version() >= 3.5:
+        p = Popen(cmd, shell=shell, timeout=timeout, stdout=PIPE, stderr=PIPE)
+        return p.communicate()
+    elif not timeout:
+        p = Popen(cmd, shell=shell, stdout=PIPE, stderr=PIPE)
+        return p.communicate()
+
+    p = Popen(cmd, shell=shell)
+    time.sleep(timeout)
+    return _, _
```

Comment:
I don't find timeout in Popen Constructor: https://docs.python.org/3.5/library/subprocess.html#subprocess.Popen

what if python < 3.5 and timeout is set? why sleep timeout